### PR TITLE
Project-Auth-Cornelia and Joel

### PR DESCRIPTION
**Pull request by JHarde**
- Frontend: https://modest-hamilton-9de3b4.netlify.app/

- Backend: https://project-auth-joel-cornelia.herokuapp.com/

### CODE REVIEW

File: frontend/src/Styling/StyledComponents.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+export const Button = styled.button`
+  width: 220px;
+  height: 50px;
+  border-radius: 30px;
+  border: 0;
+  margin: 10px;
+  font-family: "Montserrat", sans-serif;
+  background-color: #e27d60;
+  color: white;
```
Comment (Thessan) [inline frontend/src/Styling/StyledComponents.js:48]: Maybe write this line of code with HEX as well?

File: frontend/src/Styling/StyledComponents.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+export const InputField = styled.input`
+  width: 220px;
+  height: 50px;
+  border-radius: 30px;
+  border: 2px solid #e27d60;
+  text-align: left;
+  font-family: "Montserrat", sans-serif;
+  padding-left: 10px;
+  box-sizing: border-box;
+  margin: 0px;
```
Comment (Thessan) [inline frontend/src/Styling/StyledComponents.js:73]: 0; is enough 😊

File: frontend/src/Styling/StyledComponents.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  background-color: #85cdca;
+  color: #fff;
+  font-size: 17px;
+
+  ::placeholder {
+    color: #5b8f8c;
```
Comment (Thessan) [inline frontend/src/Styling/StyledComponents.js:79]: nice use of pseudo-element

File: frontend/src/Styling/StyledComponents.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  }
+`;
+
+export const StatusMessage = styled.p`
+  font-size: 20px;
+  font-weight: bold;
+  color: ${(props) => (props.success ? "#41b3a3" : "#E24030")};
```
Comment (Thessan) [inline frontend/src/Styling/StyledComponents.js:86]: great that you've used props to style the different messages 👍

File: frontend/src/user.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    setUserId: (state, action) => {
+      const { userId } = action.payload;
+      console.log(`userId: ${userId}`);
+      state.login.userId = userId;
+    },
+
+    setStatusMessage: (state, action) => {
+      const { statusMessage } = action.payload;
+      console.log(`status message: ${statusMessage}`);
```
Comment (Thessan) [inline frontend/src/user.js:29]: you can delete these console.log

**Review by Thessan — APPROVED** [review]
Great job Cornelia and Joel 🎉  You have completed blue requirements for this project. We were impressed of your usage of styled components and that you've used findOneAndUpdate method 👍  See comments for some UI suggestions.

/ Therése & Rebeka

File: backend/server.js [hunk 6]
```diff
@@ -1,29 +1,142 @@
...
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import crypto from "crypto";
+import bcrypt from "bcrypt";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+const listEndpoints = require("express-list-endpoints");
```
[the remark this answers is not in the document]
Comment (KarolCislowski) [reply backend/server.js:8]: You should add this library using the `import/from` syntax 🤓

File: frontend/src/Styling/StyledComponents.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
+import styled from "styled-components";
```
Comment (KarolCislowski) [inline frontend/src/Styling/StyledComponents.js:1]: Great idea to create the UI library for your project 🌈

File: frontend/src/user.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      state.login.statusMessage = statusMessage;
+    },
+
+    logOut: (state, action) => {
+      console.log("logging out");
+      state.login.userId = 0;
+      state.login.accessToken = null;
+    },
+  },
+});
```
Comment (KarolCislowski) [inline frontend/src/user.js:39]: Reducer is a great place to manage your users but you need to store some of your data in the local storage to keep your users logged in after they refresh the page 🤓

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
This is a very nice project 👍🏻 Great job!

Please fix the issue of unwanted logging out on page refresh and it will be an awesome app.

**Review by KarolCislowski — APPROVED** [review]
After the discussion with teachers about this assignment, we decided to approve this project without implementing the local storage functionality. 

However for the future, please remember to store your users' authentication data in your browser to make your UX good.

Good luck with your Final Project! 🍀

**Pull request closed without merging**